### ISSUE TITLE
Add a log info/warning if WAL is enabled

### ISSUE THREAD

**Issue by BTCBellyButton** [ElementsProject/lightning#5999]
### Issue
I struggled for a couple of days with crashes  ( #5984 ) to eventually find out, thanks to @grubles, that there was WAL enabled on my node and that it probably was the source of the crashes.
After disabling it my node seems to be now stable again. :crossed_fingers: 

Weird enough, I didn't even know of the existence of WAL before these issues and I have no idea how it got enabled because there was no trace of it before the upgrade from v0.12.1 to v22.12.1 in my .lightning/bitcoin directory.

### Mitigation
In order to avoid other unlucky users to stumble upon these crashes, I'd suggest to add a warning/info message at CLN startup in case WAL is enabled, alerting that it could cause random crashes and that it is safe/safer to disable it.
Instructions on how to do it would also be a nice to have in order to save some users' headaches.

**Comment by cdecker** [comment]
I'm not quite sure why WAL mode would cause crashes or other issues. Before going down this route we likely should be investigating what the root cause is and only then, if warranted, start discouraging the use of WAL.

One advantage of WAL mode is for example that the DB can be read concurrently by other processes, which I'd be rather sorry to be discouraged.

**Comment by BTCBellyButton** [comment]
I'll be happy to provide with what info you need to pinpoint the problem.
As a matter of fact, since I disabled WAL CLN didn't ever crashed.

The only "unusual" thing of my configuration is the RAID1 with mdadm.
Any idea if during the upgrade of the db WAL is enabled without the user knowing?

**Comment by BTCBellyButton** [comment]
FYI
```

udo mdadm --detail /dev/md0
[sudo] password for go: 
/dev/md0:
           Version : 1.2
     Creation Time : Wed Dec  2 13:53:24 2020
        Raid Level : raid1
        Array Size : 976630464 (931.39 GiB 1000.07 GB)
     Used Dev Size : 976630464 (931.39 GiB 1000.07 GB)
      Raid Devices : 3
     Total Devices : 3
       Persistence : Superblock is persistent

     Intent Bitmap : Internal

       Update Time : Sun Feb 12 09:46:35 2023
             State : clean 
    Active Devices : 3
   Working Devices : 3
    Failed Devices : 0
     Spare Devices : 0

Consistency Policy : bitmap

              Name : BTCPayServer:0  (local to host BTCPayServer)
              UUID : 2ef76a0c:ac5cfb7f:d5824a27:bfdf36e1
            Events : 70539

    Number   Major   Minor   RaidDevice State
       0       8       48        0      active sync   /dev/sdd
       1       8       32        1      active sync   /dev/sdc
       2       8       19        2      active sync   /dev/sdb3

```

**Comment by grubles** [comment]
FWIW on my local machine, it appears the backup database (set with `--wallet`) has WAL enabled. I think you mentioned you restored your database from a backup? Could it have been the copy created with `--wallet`? That might explain how WAL got enabled. I don't know if WAL is enabled by default for the backup.

**Comment by BTCBellyButton** [comment]
No. No backup involved. It started crashing after upgrading from 0.12.1 to  22.12.1.

**Comment by x3nb63** [comment]
are your disks SSDs or HDDs?  (spinning ones)

**Comment by BTCBellyButton** [comment]
> are your disks SSDs or HDDs? (spinning ones)

HDDs

**Comment by x3nb63** [comment]
> > are your disks SSDs or HDDs? (spinning ones)
> 
> HDDs

then your setup is like mine (in #5920), except  I use btrfs to achieve a "tripple mirror" and not mdraid (block level raid).

the HDDs come with way lower latencies - perhaps thats the actual trigger to the cause?

**Comment by BTCBellyButton** [comment]
> the HDDs come with way lower latencies - perhaps thats the trigger to the actual cause?

I don't know if that's the source of issues when Wal is active but for sure without it everything in my case appears to be back to normal.

I may be wrong but I don't see why Wal should crash with slower disks.

**Comment by x3nb63** [comment]
> I may be wrong but I don't see why Wal should crash with slower disks.

that I also do not see (while not knowing anything about its implementation). Its just a perfect stage for a race condition.

However, I used to have crashes when I had the litestream backup method configured. This one was/is based on WAL,afaik.

**Comment by BTCBellyButton** [comment]
@grubles says
> 
> Checking with a signet node, it actually appears the backup database uses the `delete` journal mode and not WAL.

**Comment by BTCBellyButton** [comment]
UPDATE

Now it is 12 consecutive days that is running without any crashes since  WAL has been disabled.

**Comment by vincenzopalazzo** [comment]
>Now it is 12 consecutive days that is running without any crashes since WAL has been disabled.

If the crash happens again, please upload the log otherwise is difficult to understand what it is going on

**Comment by BTCBellyButton** [comment]
> If the crash happens again, please upload the log otherwise is difficult to understand what it is going on

No crashes. I'd have to turn WAL on again to check...

**Comment by sangbida** [comment]
This issue is being closed as part of issue cleanup.

**Reason**: No recent activity (last updated 1074 da

If you believe this issue should remain open, please comment and we can reopen it.
